Azure Monitor: Azure Resource Graph

yesoreyeram opened the pull request:
**What this PR does / why we need it**:

Currently azure monitor datasource supports 3 different data types

* Azure monitor
* Application Insights
* Log analytics

additionally, this pull request will help to add new type of data source from **"Azure Resource Graph"**.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/17202

**Special notes for your reviewer**:

This is my first PR so feel free to suggest corrections.

@briangann For simplicity, It uses the same credentials as azure monitor. No seperate configuration settings added for **Azure Resource Graph**.  Do we need seperate configuration?

I see that we are already in the process of consolidating the configurations for different azure providers.

# Screenshots:

Sample output

![image](https://user-images.githubusercontent.com/153843/63525583-d9b41480-c4f5-11e9-8248-51ead1a233a6.png)

Query Editor

![image](https://user-images.githubusercontent.com/153843/63525476-abced000-c4f5-11e9-98ec-98f98c7898f1.png)

Using resource graph query in variables

![image](https://user-images.githubusercontent.com/153843/63316432-c8f17c00-c306-11e9-9967-fb1d5722e706.png)

yesoreyeram committed:
Azure Resource Graph

yesoreyeram committed:
Azure Resource Graph

yesoreyeram commented:
# Sample output

![image](https://user-images.githubusercontent.com/153843/63278539-5d27f880-c29f-11e9-88b9-49d514b5a354.png)

# Query Editor

![image](https://user-images.githubusercontent.com/153843/63278393-14703f80-c29f-11e9-9e13-f1bcd183868b.png)

yesoreyeram committed:
Updated docs for Resource graph

yesoreyeram committed:
Updated docs for Resource graph

yesoreyeram committed:
Options to specify number of results ( top, skip)

yesoreyeram committed:
Options to specify number of results ( top, skip)

yesoreyeram committed:
query editor converted into textarea

yesoreyeram committed:
query editor converted into textarea

yesoreyeram committed:
Resource graph query in Variables

yesoreyeram committed:
Resource graph query in Variables

yesoreyeram commented:
# Using resource graph query in variables

![image](https://user-images.githubusercontent.com/153843/63316399-a7909000-c306-11e9-96bd-6ed00b260278.png)

![image](https://user-images.githubusercontent.com/153843/63316432-c8f17c00-c306-11e9-9967-fb1d5722e706.png)

yesoreyeram committed:
test suite fix

yesoreyeram committed:
test suite fix

yesoreyeram committed:
Perf: Subscriptions cached during first run

yesoreyeram committed:
Perf: Subscriptions cached during first run

yesoreyeram committed:
Revert "Perf: Subscriptions cached during first run"

This reverts commit 07663724e8061663b08709a771492e41209a0bc6.

yesoreyeram committed:
Revert "Perf: Subscriptions cached during first run"

This reverts commit 07663724e8061663b08709a771492e41209a0bc6.

yesoreyeram committed:
Fix Subsciption cache

yesoreyeram committed:
Fix Subsciption cache

yesoreyeram committed:
Testcases added for resource graph

yesoreyeram committed:
Testcases added for resource graph

yesoreyeram committed:
Subscriptions response parser

yesoreyeram committed:
Subscriptions response parser

yesoreyeram committed:
Test coverage improved

yesoreyeram committed:
Test coverage improved

yesoreyeram committed:
Test coverage improved

yesoreyeram committed:
Test coverage improved

yesoreyeram commented:
@briangann For simplicity, It uses the same credentials as azure monitor. No seperate configuration settings added for **Azure Resource Graph**.  Do we need seperate configuration?

I see that we are already in the process of consolidating the configurations for different azure providers.

yesoreyeram committed:
testcases updated

yesoreyeram committed:
testcases updated

yesoreyeram committed:
multi target support

yesoreyeram committed:
multi target support

yesoreyeram committed:
testcases updated

yesoreyeram committed:
testcases updated

yesoreyeram committed:
Subscriptions selection

yesoreyeram committed:
Subscriptions selection

briangann commented:
We're unifying authentication methods in another PR. Separating the config (hopefully) should not be necessary.  The current config method is cumbersome.

Looking forward to see what this PR does :)

yesoreyeram committed:
Documentation update

yesoreyeram committed:
Documentation update

yesoreyeram committed:
Merge branch 'master' into azure-resourcegraph

maxilampert commented:
@briangann Any update on when this gets merged?

yesoreyeram committed:
Merge branch 'master' into azure-resourcegraph

Commit:
Merge branch 'master' into azure-resourcegraph

shavonn commented on public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_resource_graph/azure_resource_graph_datasource.ts:
```diff
@@ -0,0 +1,233 @@
+import { DataSourceInstanceSettings } from '@grafana/ui';
+import { BackendSrv } from 'app/core/services/backend_srv';
+import { TemplateSrv } from 'app/features/templating/template_srv';
+import { IQService } from 'angular';
```
We've removed the IQService from these so this should come out in the constructor and `this.$q` replace with `Promise` where ex:`return Promise.all(promises).then(results => {`

shavonn commented on public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts:
```diff
@@ -10,25 +11,36 @@ export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDa
+      instanceSettings,
+      this.backendSrv,
+      this.templateSrv,
+      this.$q
```
Remove as arg.

shavonn reviewed (changes requested):
Just these couple of changes and fix tests.

ryantxu committed:
Toolkit: copyIfNonExistent order swapped (#21653)

torkelo committed:
Alert: Minor tweak to work with license warnings (#21654)

hugohaggmark committed:
Chore: Fixes PhantomJs by adding polyfills for fetch and AbortController (#21655)

* Chore: Fixes PhantomJs rendering

* Docs: Updated docs

* Update contribute/style-guides/e2e.md

Co-Authored-By: Dominik Prokop <dominik.prokop@grafana.com>

* Docs: Updates docs according to PR comments

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>

torkelo committed:
Login: Better auto sizing of login logo (#21645)

sunker committed:
CloudWatch: Auto period snap to next higher period (#21659)

* Snap to next higher period instead of closest

* Adjust period calc

torkelo committed:
Docs: Whats new updates (#21664)

* Docs: Whats new updates

* Add cloudwatch and stackdriver paragraphs

* Minor changes

* Add link helper to render gif

* Docs: what's new updates

Co-authored-by: Erik Sundell <erik.sundell87@gmail.com>

dprokop committed:
Migrations: migrate admin user create page (#21506)

* Create basic react page for AdminUserCreate

* Forms.Field - render asterisk when field is marked as required

* Add validation to user create form wih react-hook-form

* Remove Angular code for UserCreatePage

* Remove commented route for admin settings

* Remove unused import

* Hide react-hooks-form behind Form component

* Fix webkit autofill

* Webkit autofill on inpiuts - bring back focus shadow

* Temporarily fix story (before 21635 is merged)

* Form: docs and minor updates to new form elements (#21635)

* Allow Switch, checkbox to be uncontrolled, forward refs, styles update

* Add Form docs

* User create page update

* Remove unused import

* Apply review notes

papagian committed:
SQLStore: Fix PostgreSQL failure to create organisation for first time (#21648)

* Fix PostgreSQL failure to create organisation for first time

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

kaydelaney committed:
Explore: Fixes some LogDetailsRow markup (#21671)

- Moves filter titles to icons rather than table cell
- Increases colspan of ad-hoc stats cell instead of
rendering empty cells for parsed fields

sunker committed:
Run query when region, namespace and metric changes (#21633)

dprokop committed:
Circle: Introduce es-check to branches & pr workflow  (#21677)

* Transpile selected es6 node deps

* Add es-check to build-fast-frontend

* change es-check is used

* transpile react-hook-form

torkelo committed:
StatPanel: minor height tweak (#21663)

papagian committed:
Update changelog generation to ignore not merged pull requests (#21641)

kaydelaney committed:
backendSrv: Only stringifies request body if payload isn't already a string (#21639)

* backendSrv: Only stringifies request body if payload isn't already a string

* Add test cases to verify behaviour

dprokop committed:
PhantomJS: Fix rendering of panels using Prometheus datasource

In 043bb59 a URLSearchParams usage was introduced which is not supported by PhantomJS. @babel/polyfill(deprecated) does not contain polyfill for URLSearchParams, hence the code (and Prometheus graphs rendering) was failing in PhantomJS environment.

The solution is to add https://www.npmjs.com/package/url-search-params-polyfill that takes care of the URLSearchParams

johntdyer committed:
Alerting: Support passing tags to Pagerduty and allow notification on specific event categories (#21335)

Add support for passing tags to Pagerduty and allow notification 
on specific event categories such as Class, Group and Component.

Ref #19912, #19913

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

marefr committed:
Docs: Fix broken link in upgrade notes (#21698)

Fixes a broken link in the 6.6 upgrade notes.

papagian committed:
Docs: Update what's new in 6.6 (#21699)

* Update what's new in 6.6

* Add upgrading and changelog sections

* Remove lines

* Update cookie management section

* Update Bar Gauge unfilled option

* Add missing dot

Commit:
Make importDataSourcePlugin cancelable (#21430)

* make importDataSourcePlugin cancelable

* fix imported plugin assignment

* init datasource plugin to redux

* remove commented

* testDataSource to redux

* add err console log

* isTesting is never used

* tests, loadError type

* more tests, testingStatus obj

torkelo committed:
StatPanels: Fixed possible migration issue (#21681)

aknuds1 committed:
CircleCI: Add workflow for building with Grafana Build Pipeline (#21449)

* CircleCI: Add workflow for building with grabpl

ivanahuckova committed:
Loki: Fix Loki with repeated panels and interpolation for Explore (#21685)

dprokop committed:
grafana/ui: Remove path import  from grafana-data (#21707)

dprokop committed:
Annotations: Fix issue with annotation queries editors (#21712)

marefr committed:
Admin: Viewer should not see link to teams in side menu (#21716)

Fixes so that viewers don't see a link to teams in side menu when
editors_can_admin setting is enabled.

torkelo committed:
Footer: added back missing footer to login page (#21720)

torkelo committed:
Devenv: Fixed devenv dashboard template var datasource (#21715)

aocenas committed:
Loki: Refactor editor and syntax hooks (#21687)

Commit:
Template vars: Add error message for failed query var (#21731)

xlson committed:
Docs: Enterprise 6.6 (#21666)

* Docs: gitlab team-sync documented correctly

* docs: initial docs for whitelabeling

* Docs: enterprise repositories

* Docs: rpm enterprise install

* Docs: re-structured the rpm install page to be more similar to the deb page

* Docs: responded to review feedback

* Docs: recommends the enterprise edition

* Update docs/sources/installation/debian.md

yesoreyeram commented:
Sure @shavonn will update this next week.

aknuds1 committed:
Docs: Fix getting started links on Windows installation page (#21724)

tskarhed committed:
TablePabel: Sanitize column link (#21735)

ryantxu committed:
Toolkit: add canvas-mock to test setup (#21739)

yesoreyeram committed:
Merge branch 'azure-resourcegraph' of https://github.com/yesoreyeram/grafana into azure-resourcegraph

yesoreyeram committed:
rebase and comments resolved

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=18629) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=18629) before we can accept your contribution.<br/>**16** out of **17** committers have signed the CLA.<br/><br/>:white_check_mark: ryantxu<br/>:white_check_mark: kaydelaney<br/>:white_check_mark: torkelo<br/>:white_check_mark: yesoreyeram<br/>:white_check_mark: shavonn<br/>:white_check_mark: hugohaggmark<br/>:white_check_mark: aocenas<br/>:white_check_mark: papagian<br/>:white_check_mark: sunker<br/>:white_check_mark: dprokop<br/>:white_check_mark: johntdyer<br/>:white_check_mark: ivanahuckova<br/>:white_check_mark: aknuds1<br/>:white_check_mark: marefr<br/>:white_check_mark: tskarhed<br/>:white_check_mark: xlson<br/>:x: Tobias Skarhed<br/><hr/>**Tobias Skarhed** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=18629) it.</sub>

yesoreyeram committed:
Removed backendSrv dependency

yesoreyeram commented:
Git rebase messed up my PR with many changes happened in the master meanwhile. I am working on cleanup and making changes to align with master.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=18629) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=18629) before we can accept your contribution.<br/>**16** out of **17** committers have signed the CLA.<br/><br/>:white_check_mark: shavonn<br/>:white_check_mark: hugohaggmark<br/>:white_check_mark: ryantxu<br/>:white_check_mark: yesoreyeram<br/>:white_check_mark: dprokop<br/>:white_check_mark: johntdyer<br/>:white_check_mark: torkelo<br/>:white_check_mark: papagian<br/>:white_check_mark: aknuds1<br/>:white_check_mark: sunker<br/>:white_check_mark: kaydelaney<br/>:white_check_mark: ivanahuckova<br/>:white_check_mark: tskarhed<br/>:white_check_mark: marefr<br/>:white_check_mark: aocenas<br/>:white_check_mark: xlson<br/>:x: Tobias Skarhed<br/><hr/>**Tobias Skarhed** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=18629) it.</sub>

bergquist commented:
Moving this to 7.0 since this PR is not ready for review.
